Unit test

aposdroid opened the pull request.

aposdroid committed:
add jacoco report

aposdroid committed:
Исправил класс Lion, создал 4 тестовых класса

digranate commented on pom.xml:
```diff
@@ -8,6 +8,22 @@
     <artifactId>untitled</artifactId>
     <version>1.0-SNAPSHOT</version>
 
+    <dependencies>
```
Файлы .idea коммитить не нужно. Лучше сразу добавить их в gitignore.

aposdroid replied:
Добавил

digranate commented on src/main/java/com/example/Lion.java:
```diff
@@ -6,7 +6,8 @@ public class Lion {
     boolean hasMane;
 
-    public Lion(String sex) throws Exception {
+    public Lion(String sex, Feline feline) throws Exception {
```
Нужно ещё убрать создание нового объекта на поле feline. Это лишняя трата ресурсов.

aposdroid replied:
Убрал

digranate commented on src/test/java/com/example/CatTest.java:
```diff
@@ -0,0 +1,36 @@
+
+    public void catGetFoodTest() throws Exception {
+        Cat cat = new Cat(feline);
+        Mockito.when(cat.getFood()).thenReturn( List.of("Животные", "Птицы", "Рыба"));
```
Мы никогда не подменяем результат того метода, который сейчас тестируем. Подменяем только зависимости.

aposdroid replied:
Исправил

digranate commented on src/test/java/com/example/CatTest.java:
```diff
@@ -0,0 +1,36 @@
+        Cat cat = new Cat(feline);
+        Mockito.when(cat.getFood()).thenReturn( List.of("Животные", "Птицы", "Рыба"));
+
+        Assert.assertEquals((List.of("Животные", "Птицы", "Рыба")), cat.getFood());
```
Дважды создаёшь один и тот же список. Лучше переиспользовать.

aposdroid replied:
Создал переменную и переиспользовал

digranate commented on src/test/java/com/example/LionManeExceptionTest.java:
```diff
@@ -0,0 +1,4 @@
+package com.example;
+
+public class LionsManeExceptionTest {
+}
```
Интересный класс)

aposdroid replied:
Это я пытался в него запихнуть проверку вывода ошибки, но не придумал как

aposdroid replied:
а щас придумал

digranate replied:
Супер!

pomegranate282922 reviewed (commented):
Хорошая работа! Пара важных замечаний есть - прошу их поправить.

aposdroid committed:
Добавил .idea в gitignore

aposdroid committed:
Добавил .idea в gitignore.
Убрал создание лишнего объекта feline в классе Lion.

aposdroid committed:
Добавил .idea в gitignore.
Убрал создание лишнего объекта feline в классе Lion.
Исправил подмену результата в тесте catGetFoodTest.

aposdroid committed:
Добавил .idea в gitignore.
Убрал создание лишнего объекта feline в классе Lion.
Исправил подмену результата в тесте catGetFoodTest.
Переиспользовал список в методе catGetFoodTest.

aposdroid committed:
Добавил .idea/ в gitignore.
Убрал создание лишнего объекта feline в классе Lion.
Исправил подмену результата в тесте catGetFoodTest.
Переиспользовал список в методе catGetFoodTest.
Добавил тест LionsManeExceptionTest, который проверяет, что выводится ошибка при неправильном аргументе.

digranate commented on src/test/java/com/example/CatTest.java:
```diff
@@ -0,0 +1,37 @@
+    Feline feline;
+
+    @Test
+
```
Лишняя строка

pomegranate282922 reviewed (approved):
Зачёт!
